src/components/messagebox/ChattingRoom/index.tsx

```diff
@@ -1,114 +1,150 @@
-import { Image, IconButton, Input } from '@offer-ui/react'
+import { Image, IconButton, Input, useMedia } from '@offer-ui/react'
+import { useEffect, useRef, useState } from 'react'
 import { Styled } from './styled'
 import type { ChattingRoomProps } from './types'
 import { Chatting } from '../Chatting'
+import type { ChattingProps } from '../Chatting/types'
+import { Dialog } from '@components/common'
 import { toLocaleCurrency } from '@utils/format'
+import {
+  useCreateMessageMutation,
+  useDeleteMessageRoomMutation,
+  useGetMessageQuery
+} from '@apis'
+import { useAuth, useModal } from '@hooks'
 
-export const ChattingRoom = ({ id, onClose }: ChattingRoomProps) => {
-  // TODO: ChattingRoom Id를 통해서 컴포넌트 내부에서 패치하도록
+// TODO: messageRoom 정보조회 api 붙이고 제거
+const POST_MOCK = {
+  offerPrice: 10000,
+  post: {
+    title: '팔아요 !',
+    thumbnailImageUrl: '',
+    price: 12000
+  }
+}
+
+export const ChattingRoom = ({ roomId, onClose }: ChattingRoomProps) => {
+  const getMessageQuery = useGetMessageQuery({
+    msgRoomId: roomId,
+    page: 0
+  })
+  const createMessageMutation = useCreateMessageMutation(roomId)
+  const deleteMessageRoomMutation = useDeleteMessageRoomMutation(roomId)
+  const chattingBoxRef = useRef<HTMLDivElement | null>(null)
+  const inputRef = useRef<HTMLInputElement | null>(null)
+  const { desktop, tablet, mobile } = useMedia()
+  const [messages, setMessages] = useState<ChattingProps['messages']>([])
+  const { isOpen, openModal, closeModal } = useModal()
+  const { user } = useAuth()
+  const senderInfo = {
+    id: user.id,
+    nickname: user.nickname,
+    imageUrl: user.profileImageUrl
+  }
```

Comment:
```suggestion
  const createMessageMutation = useCreateMessageMutation(roomId)
  const deleteMessageRoomMutation = useDeleteMessageRoomMutation(roomId)
  
  const chattingBoxRef = useRef<HTMLDivElement | null>(null)
  const inputRef = useRef<HTMLInputElement | null>(null)
  
  const [messages, setMessages] = useState<ChattingProps['messages']>([])
  const { desktop, tablet, mobile } = useMedia()
  const { isOpen, openModal, closeModal } = useModal()
  const { user } = useAuth()
  
  const senderInfo = {
    id: user.id,
    nickname: user.nickname,
    imageUrl: user.profileImageUrl
  }
```
custom hook, state 같이 붙어있는게 더 읽기 편할 것 같아 위치와 간격만 수정해보았슴니다!

Reply:
저도 분류하는 것이 가독성면에서 훨씬 좋다고 생각해요!
분류하다보니 종종 순서를 섞어야할 때도 있더라구요 (예를 들면 state가 먼저 선언돼서 query의 param으로 들어가야 한다던지!)
그래서 분류 의미가 조금 무색해지는 느낌이 있어 고민했었습니다 🤓

[변경 완료 ! ](https://github.com/price-offer/offer-fe/pull/196/commits/c334aa356693e396051ebe90551f92c71661cf30#diff-2e1f52de9ff48c1e322340a310c75b772aa52c734108dbf9888ba96a88db7ce3R31-R41)